### PR TITLE
Custom loggers

### PR DESCRIPTION
**Pull request by barberj**
Enable adding custom loggers to available formatters.

### CODE REVIEW

**Comment by barberj** [comment]
@jnunemaker thoughts?

File: lib/httparty/logger/logger.rb [hunk 1]
```diff
@@ -3,16 +3,23 @@
 
 module HTTParty
   module Logger
+    def self.formatters
```
Comment (jnunemaker) [inline lib/httparty/logger/logger.rb]: Thoughts on calling this loggers instead of formatters?

File: lib/httparty/logger/logger.rb [hunk 2]
```diff
@@ -3,16 +3,23 @@
...
+      @formatters ||= {
+        :curl => Logger::CurlLogger,
+        :apache => Logger::ApacheLogger
+      }
+    end
+
+    def self.add_formatters custom_formatters
```
Comment (jnunemaker) [inline lib/httparty/logger/logger.rb]: I'd probably singularize this `add_logger` or `add_formater` and only add one at a time. What should the behavior be if you provide a key that already exists? This seems to just allow overriding which could be confusing. I'm wondering if we should raise if the logger already is defined.

File: lib/httparty/logger/logger.rb (hunk 1 above)
Comment (jnunemaker) [inline lib/httparty/logger/logger.rb]: Actually, I'm ok with formatters. I as I look over how it works I see that these take a logger and format requests and responses so this works.

**Comment by barberj** [comment]
@jnunemaker hows this?

I almost feel `CurlLogger` and `ApacheLogger` should be renamed to `CurlFormatter` `ApacheFormatter`

**Comment by jnunemaker** [comment]
> I almost feel CurlLogger and ApacheLogger should be renamed to CurlFormatter ApacheFormatter

Agreed. I'll merge this. Feel free to whip another PR for that if you have time. If not, we'll get to it eventually.

**Pull request merged**